pallets/oracle/src/lib.rs

```diff
@@ -211,60 +162,20 @@ pub mod pallet {
 	// The pallet's dispatchable functions.
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
-		/// Feeds data from the oracles, e.g., the exchange rates. This function
-		/// is intended to be API-compatible with orml-oracle.
-		///
-		/// # Arguments
-		///
-		/// * `values` - a vector of (key, value) pairs to submit
-		#[pallet::call_index(0)]
-		#[pallet::weight(<T as Config>::WeightInfo::feed_values(values.len() as u32))]
-		pub fn feed_values(
-			origin: OriginFor<T>,
-			values: Vec<(OracleKey, T::UnsignedFixedPoint)>,
-		) -> DispatchResultWithPostInfo {
-			let signer = ensure_signed(origin)?;
-
-			// fail if the signer is not an authorized oracle
-			ensure!(Self::is_authorized(&signer), Error::<T>::InvalidOracleSource);
-
-			Self::_feed_values(signer, values);
-			Ok(Pays::No.into())
-		}
-
-		/// Adds an authorized oracle account (only executable by the Root account)
-		///
-		/// # Arguments
-		/// * `account_id` - the account Id of the oracle
-		/// * `name` - a descriptive name for the oracle
-		#[pallet::call_index(1)]
-		#[pallet::weight(<T as Config>::WeightInfo::insert_authorized_oracle())]
-		#[transactional]
-		pub fn insert_authorized_oracle(
-			origin: OriginFor<T>,
-			account_id: T::AccountId,
-			name: Vec<u8>,
-		) -> DispatchResult {
-			ensure_root(origin)?;
-			Self::insert_oracle(account_id.clone(), name.clone());
-			Self::deposit_event(Event::OracleAdded { oracle_id: account_id, name });
-			Ok(())
-		}
-
-		/// Removes an authorized oracle account (only executable by the Root account)
+		/// set oracle keys
 		///
 		/// # Arguments
-		/// * `account_id` - the account Id of the oracle
+		/// * `oracle_key` - list of oracle keys
 		#[pallet::call_index(2)]
 		#[pallet::weight(<T as Config>::WeightInfo::remove_authorized_oracle())]
 		#[transactional]
-		pub fn remove_authorized_oracle(
+		pub fn update_oracle_key(
 			origin: OriginFor<T>,
-			account_id: T::AccountId,
+			oracle_key: Vec<OracleKey>,
```

Comment:
This should be named `update_oracle_keys()` and the included variable should be `oracle_keys` since we use a vec here.